app/importers/rows/attendance_row.rb

```diff
@@ -16,6 +16,8 @@ def save; end
       def valid?; false end
 
       def errors; NullEventErrors.new end
+
+      def assign_attributes(attributes); end
```

Comment:
This seems leftover, or am I missing what this is doing?

Reply:
This is because this PR uses `assign_attributes` in AttendanceRow. My mental model is that "reason" and "comment" are fields that might get updated after the fact. I don't want to treat absences for the same student on the same day with different reasons or comments as different events. NullRelation needs to have all of the same method names as AttendanceRow to follow the null object pattern. This is how we handle events that aren't Absences or Tardies through duck typing.

Reply:
Ah, thanks!  That's super helpful. 👍 I hear what you're saying describing how `NullRelation` works, and if I look back at [attendance_importer.rb](https://github.com/studentinsights/studentinsights/blob/68252fea01b64e8110f72ea73d9f5a648e0d83c8/app/importers/file_importers/attendance_importer.rb#L59) I see the other piece of what you're saying.

This is probably a discussion for a separate time, but I'll  think using the null object pattern in this situation is tricky, since AR models have a huge interface that you'd have to recreate.  Reading how the `NullRelation` gets created here, it looks like it's used to signal if the row isn't flagged as an absence or a tardy  I'm wondering if there are other ways to approach this, maybe having the `#build` method return exit early and return `nil` to indicate that case.  Or to have the `#build` method return either or AR object or an error object, and let the caller decide what to do rather than trying to hide that from the caller altogether.

And to be explicit, I don't think this is something we should tackle in this PR, just sharing as it comes up and maybe it's something we do later separately.

Reply:
@kevinrobinson Yeah, I agree with that analysis, I don't think null object pattern makes the most sense here either.